notation.go

```diff
@@ -0,0 +1,73 @@
+package notation
+
+import (
+	"context"
+	"time"
+
+	"github.com/opencontainers/go-digest"
+)
+
+// Descriptor describes the content signed or to be signed.
+type Descriptor struct {
+	// MediaType is the media type of the targeted content.
+	MediaType string `json:"mediaType"`
+
+	// Digest is the digest of the targeted content.
+	Digest digest.Digest `json:"digest"`
+
+	// Size specifies the size in bytes of the blob.
+	Size int64 `json:"size"`
+}
+
+// Metadata contains informational metadata about the signed artifact.
+type Metadata struct {
+	// Identity describes the artifact identity.
+	Identity string
+
+	// Attributes contains user defined attributes.
+	Attributes map[string]interface{}
+}
+
+// SignOptions contains parameters for Signer.Sign.
+type SignOptions struct {
```

Comment:
Timestamping is currently a per signer operation. Should we make it optional as per signature signing?

Reply:
One signer implementation(like file-based one) can be reused by multiple organization. Some organizations might not want to timestamp signature because of availability concerns or something else but other might prefer to timestamp signature.
So there should be a way to for user to disable/enable time-stamping on per signature basis. One way to achieve is that only add timestamp when user explicitly specifies it
* notation sign --timestamp=${serverURL} <-- signs and timestamps
* notation sign  <-- only signs